bin/newman.js

```diff
@@ -2,11 +2,261 @@
 require('../lib/node-version-check'); // @note that this should not respect CLI --silent
 
 var _ = require('lodash'),
-
-    cli = require('../lib/cli'),
+    program = require('commander'),
+    version = require('../package.json').version,
     newman = require('../'),
 
     /**
+     * A CLI parser helper to process stringified numbers into integers, perform safety checks, and return the result.
+     *
+     * @param {String} arg - The stringified number argument pulled from the CLI arguments list.
+     * @returns {Number} - The supplied argument, casted to an integer.
+     */
+    Integer = function (arg) {
+        var num = Number(arg);
+
+        if (!_.isSafeInteger(num) || num <= 0) {
+            throw new Error('The value must be a positive integer.');
+        }
+
+        return num.valueOf();
+    },
+
+    /**
+     * Separates reporter specific arguments from the rest.
+     *
+     * @todo Figure out a better way! (Or submit a PR to argparse).
+     * @param {Array} allArgs - An array of strings, each coresponding to a CLI argument.
+     * @returns {Object} - An object with reporter and regular argument key-values.
+     */
+    separateReporterArgs = function (allArgs) {
+        var reporterArgs = [],
+            args = [],
+            i;
+
+        // Separate the reporter arguments from the rest
+        for (i = 0; i < allArgs.length; i++) {
+            if (_.startsWith(allArgs[i], '--reporter-')) {
+                reporterArgs.push(allArgs[i]);
+
+                // Also push the next parameter if it's not an option.
+                allArgs[i + 1] && !_.startsWith(allArgs[i + 1], '-') && reporterArgs.push(allArgs[++i]);
+            }
+            else {
+                args.push(allArgs[i]);
+            }
+        }
+        return {
+            reporter: reporterArgs,
+            argv: args
+        };
+    },
+
+    /**
+     *  used for collecting global key=value variables supplied through command line
+     *
+     * --global-var "foo=bar" --global-var "alpha=beta"
+     *
+     * @param {String} val - The argument provided to `--global-var`.
+     * @param {String} memo - The array that is populated by key value pairs.
+     * @returns {{key, value}} - The object representation of the current CLI variable.
+    */
+    collect = function (val, memo) {
+        var arg = val,
+            eqIndex = arg.indexOf('='),
```

Comment:
we can avoid changing the datatype of `arg` by
```js
var arg,
   eqIndex = val.indexOf('='),
   arg = ... ;
```

Reply:
Noted.